src/bucketizers/index.ts

```diff
@@ -139,12 +151,24 @@ export class BucketizerOrchestrator {
                 const bucketizer = this.getBucketizer(i, prefix);
 
                 const getBucket = (value: string, root?: boolean) => {
-                    const terms = value.split("/");
-                    const key = encodeURIComponent(
-                        decodeURIComponent(terms[terms.length - 1]),
-                    );
+                    const encodedValue = value
+                        .split("/")
+                        .map((x) => encodeURIComponent(decodeURIComponent(x)))
+                        .join("/");
+                    const key = value.endsWith("/")
+                        ? encodedValue
+                        : encodedValue + "/";
+                    // const key = value;
                     // If the requested bucket is the root, it actually is the previous bucket
-                    const id = root ? prefix : prefix + "/" + key;
+
+                    // avoid double slashes and leading slashes
+                    const next =
+                        prefix.endsWith("/") ||
```

Comment:
What if both the prefix ends with a slash, and the key starts with a slash?

Reply:
:exploding_head: